src/wallet/wallet.h

```diff
@@ -1080,6 +1080,9 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     /** Mark a transaction as replaced by another transaction (e.g., BIP 125). */
     bool MarkReplaced(const uint256& originalHash, const uint256& newHash);
 
+    //! Verify wallet naming and perform salvage on the wallet if required
+    static bool VerifyWallet(std::string walletFile, bool salvage_wallet);
```

Comment:
@TheBlueMatt btw my suggestion is to move (some) static code to a `WalletManager` (#13034).

Reply:
I agree with @promag that a WalletManager class is a good place for this in the long run.

Since this is a nit only, can we save it for a future PR.